File: 2279-maximum-bags-with-full-capacity-of-rocks/2279-maximum-bags-with-full-capacity-of-rocks.cpp
class Solution {
public:
    int maximumBags(vector<int>& c, vector<int>& r, int a) {
        int cnt = 0;
        vector<pair<int, int>> v;
        int n = c.size();
        v.resize(n);
        for (int i = 0; i < n; ++i)
        {
            v[i] = make_pair(c[i], r[i]);
        }
        sort(v.begin(), v.end(), [&](pair<int, int> a, pair<int, int> b){
            return a.first - a.second < b.first - b.second;
        });
        for (int i = 0; i < n; ++i)
        {
            if (v[i].first - v[i].second > 0)
            {
                a -= v[i].first - v[i].second;
            }
            if (a < 0)
            {
                break;
            }
            cnt++;
        }
        return cnt;
    }
};

File: 2279-maximum-bags-with-full-capacity-of-rocks/2279-maximum-bags-with-full-capacity-of-rocks.py
class Solution:
    def maximumBags(self, capacity: List[int], rocks: List[int], additionalRocks: int) -> int:
        bag = []
        n = len(capacity)
        
        for i in range(n):
            bag.append((capacity[i], rocks[i]))
            
        bag.sort(key = lambda x: x[0] - x[1])
        
        cnt = 0
        
        for i in range(n):
            deviant = bag[i][0] - bag[i][1]
            if deviant > 0:
                additionalRocks -= deviant
            if additionalRocks < 0:
                break
            cnt += 1
        
        return cnt